src/router/index.js

```diff
@@ -38,8 +40,27 @@ let router = new Router({
     },
     {
       path: '/profile',
-      name: 'Home',
+      name: 'Profile',
       component: Profile,
+      meta: {
+        requiresAuth: true,
+      },
+      children: [
+        { path: ':id', component: ProfileDetail },
+        {
+          path: ':id/edit',
+          component: ProfileEdit,
+          beforeEnter (to, from, next) {
+            const routeId = to.params.id
+            const currentUser = Auth.auth.currentUser
+            if (routeId !== currentUser.uid) {
+              next(`/profile/${routeId}`)
+            } else {
+              next()
```

Comment:
so i'm a bit of a fan of returning next() type middleware stuff even though you don't need the return value, i think it reads a bit better than if/else.  The returns just serve to break the function 

also, where is `Auth` coming from?